flask-server/tests/test_user_api.py

```diff
@@ -0,0 +1,267 @@
+import unittest
+from test_api import TestAPI
+from routes.user import check_email
+from models.user_models import User
+
+
+class TestUserAPI(TestAPI):
+    """Tests for user namespace routes and related functions"""
+
+    def test_valid_email(self):
+        """Test a valid email address with check email"""
+        result = check_email("test@email.com")
+        self.assertTrue(result)
+
+    def test_invalid_email(self):
+        """Test an invalid email address with check email"""
+        result = check_email("test.com")
+        self.assertFalse(result)
+
+    def test_register_successful(self):
+        """Test successful registration of user"""
+        register_response = self.client.post("/user/register",
+            json = {
+                "username": "testuser",
+                "first_name": "test",
+                "last_name": "user",
+                "email": "testuser@test.com",
+                "password": "mytestpassword"
+            }
+        )
+
+        status_code = register_response.status_code
+
+        self.assertEqual(status_code, 201)
+
+    def test_register_successful_max_length(self):
+        """Test successful registration of user, username, first name and last name max characters"""
```

Comment:
Part of your registration logic is to login the user and return it's token right?
It would be nice to fetch that token here as well and validate that it's filled and works for authentication.

Reply:
Will look into this more and look into different methods of authentication as well for something more secure.